crates/signal/src/lib.rs

```diff
@@ -65,15 +63,128 @@
 use std::pin::Pin;
 use std::task::{Context, Poll};
 
-use tokio::signal::unix::{Signal, SignalKind};
+#[cfg(unix)]
+use tokio::signal::unix::{Signal as UnixSignal, SignalKind as UnixSignalKind};
+#[cfg(windows)]
+use tokio::signal::windows;
 
 #[cfg(feature = "bootstrap")]
 mod bootstrap;
 
 #[cfg(feature = "bootstrap")]
 pub use bootstrap::{SignalConfig, SignalSvc};
 
-/// A handler for listening to multiple Unix signals, and providing a future for
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum SignalKind {
+    Interrupt,
+    Terminate,
+    #[cfg(windows)]
+    Windows(WindowsSignalKind),
+    #[cfg(unix)]
+    Unix(UnixSignalKind),
+}
+
+#[cfg(unix)]
+impl From<UnixSignalKind> for SignalKind {
+  fn from(value: UnixSignalKind) -> Self {
+    Self::Unix(value)
+  }
+}
+
+#[cfg(unix)]
+impl PartialEq<UnixSignalKind> for SignalKind {
+    fn eq(&self, other: &UnixSignalKind) -> bool {
+        match self {
+            Self::Unix(kind) => kind == other,
+            _ => false,
+        }
+    }
+}
```

Comment:
i think we should also take into account, partial equals for the interrupt / term being compared to unix interrupt / term.